go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -575,14 +638,67 @@ func (app *registryApplication) registerRuntime( // nolint: gocyclo
 		}
 	}
 
-	// Make sure that the entity has enough stake.
-	if !params.DebugBypassStake {
+	if !ctx.IsInitChain() {
+		// Make sure the signer of the transaction matches the signer of the
+		// entity or runtime that is controlling the runtime.
+		// NOTE: If this is invoked during InitChain then there is no actual transaction
+		//       and thus no transaction signer so we must skip this check.
+
+		// If we're updating the governance model, we should check the signer
+		// based on the existing governance model and not the new one,
+		// otherwise it would be impossible to transition from entity to
+		// runtime governance, for example.
+		var rtToCheck *registry.Runtime
+		if existingRt != nil {
+			rtToCheck = existingRt
+		} else {
+			rtToCheck = rt
+		}
+
+		switch rtToCheck.GovernanceModel {
+		case registry.GovernanceEntity:
+			if !rt.EntityID.Equal(ctx.TxSigner()) {
```

Comment:
I know the EntityID cannot change but this makes it more clear IMO:
```suggestion
			if !rtToCheck.EntityID.Equal(ctx.TxSigner()) {
```